### PR TITLE
new(libsinsp/test): add spawn process test

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area libsinsp
/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

This PR adds a convenience function plus the first test that checks process spawn tracking. This increases libsinsp coverage from ~17% to ~19% . This also should serve as a simple case that can be made more complex to test more edge cases.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
I love this! Even if CI is somewhat stalled here :/

File: userspace/libsinsp/test/test_utils.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+*/
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace test_utils {
+
+std::string to_null_delimited(std::vector<std::string> list);
```
Comment (Andreagit97) [inline userspace/libsinsp/test/test_utils.h:25]: If you have time I will add a little comment to let people understand that you put a terminator after every single string and not only at the end. Obviously it is clear from the code but not so much from the name... just one line to put lazy developers in the right direction is good for me! This is a minor so fill free to ignore it :)

**Review by Andreagit97 — APPROVED** [review]
Left just a minor comment but feel free to ignore it if it sounds stupid. By the way amazing work!

**Review by FedeDP — APPROVED** [review]
/approve

File: userspace/libsinsp/test/test_utils.h (hunk 1 above)
Comment (LucaGuerra) [reply userspace/libsinsp/test/test_utils.h:27]: Sure, comment added!

**Review by FedeDP — APPROVED** [review]
/approve

**Review by leogr — APPROVED** [review]

**Pull request merged**